src/pages/index.js

```diff
@@ -0,0 +1,319 @@
+const profileImageSubmitBtn = profileEditImageFormElement.querySelector(
+  ".modal__submit-btn--edit-avatar-profile"
+);
+
+// New Post Modal
+const newPostModal = document.querySelector("#new-post-modal");
+const newPostBtn = document.querySelector(".profile__add-btn");
+const newPostCloseBtn = document.querySelector(".modal__close-btn--new-post");
+const newPostSubmitBtn = document.querySelector(".modal__submit-btn--new-post");
+const newPostFormElement = document.querySelector(".modal__form--new-post");
+const imageLinkInput = newPostFormElement.querySelector("#image-link");
+const captionInput = newPostFormElement.querySelector("#post-caption");
+
+// Delete Post Modal
+const deletePostModal = document.querySelector("#delete-card-modal");
+const deletePostBtn = document.querySelector(".modal__card-delete-btn");
+const canelDeletePostBtn = document.querySelector(
+  ".modal__card-delete-cancel-btn"
+);
+const deleteModalCloseBtn = document.querySelector(
+  ".modal__close-btn--delete-card"
+);
+
+// Preview Modal
+const previewModal = document.querySelector("#preview-modal");
+const previewImage = previewModal.querySelector(".modal__image--preview");
+const previewCaption = previewModal.querySelector(".modal__caption--preview");
+const previewCloseBtn = previewModal.querySelector(
+  ".modal__close-btn--preview"
+);
+
+// Cards
+const cardsList = document.querySelector(".cards__list");
+
+// === Fetch User Info ===
+api
+  .getUserInfo()
+  .then((data) => {
+    profileNameElement.textContent = data.name;
+    profileJobElement.textContent = data.about;
+    profileImage.src = data.avatar;
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+// === Fetch and Render Cards ===
+api
+  .getInitialCards()
+  .then((data) => data.map((item) => item))
+  .then((cards) => {
+    cards.forEach((card) => {
+      const newCard = getCardElement(card);
+      cardsList.append(newCard);
+    });
+  })
+  .catch(console.error);
+
+// === Card Logic ===
+let selectedCard;
+let selectedCardID;
+
+function getCardElement(data) {
+  const cardTemplate = document.querySelector("#card__template").content;
+  const cardElement = cardTemplate.querySelector(".card").cloneNode(true);
+  const cardTitle = cardElement.querySelector(".card__title");
+  const cardImage = cardElement.querySelector(".card__image");
+  const cardLikeBtn = cardElement.querySelector(".card__like-btn");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
+
+  cardTitle.textContent = data.name;
+  cardImage.src = data.link;
+  cardImage.alt = data.name;
+
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-btn_liked");
+  }
+
+  cardImage.addEventListener("click", () => {
+    previewImage.src = data.link;
+    previewImage.alt = data.name;
+    previewCaption.textContent = data.name;
+    openModal(previewModal);
+  });
+
+  cardLikeBtn.addEventListener("click", () => {
+    cardLikeBtn.classList.toggle("card__like-btn_liked");
```

Comment:
You need to **toggle likes** only in a `then` block after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.